ivy/functional/backends/numpy/experimental/sorting.py

```diff
@@ -11,3 +11,14 @@ def msort(
 
 
 msort_support_native_out = False
+
+
+# lexsort
+def lexsort(
```

Comment:
So I suppose you should have one of this lexsort definition in each of the backends/[frameworks]/experimencal/sorting.py file

Reply:
I added lexsort definition in each of the backends but it is not completed because for instance tensorflow has no tf.lexsort. I will implement also lexsort functions in ivy>functional>backends>[frameworks]>sorting.py

Reply:
As tensorflow does not have a lexsort for you to use, why not try to recreate the expected behavior through a combination of multiple tensorflow functions? Like adding some extra steps after using a function of tensorflow that does a similar job like lexsort?

Reply:
I am thinking this is the best case for lexsort torch implementation compared to the other cases I tried. But I am getting an error(AssertionError: [0 1] != 0)😬​, do you have any suggestions for me?

Reply:
Can I have a more detailed example of that assertion error?

Reply:
Yes, you are right, I am sorry. This is the all [test results of failed tests](https://drive.google.com/file/d/14THp9dnPYcrdYcoyoHKfBGfFqn8Petme/view?usp=sharing). This is 16 pages document. I marked the assertion error part(13.,14. pgs).

Reply:
I also rarely did well for the torch test, I don't think I've taken a probabilistic step. I'm wrong about one thing, that's for sure

Reply:
At first glance, it seems like your function returns a list of two numbers while the real counter part from numpy returns only 0. Then I searched about how the lexsort behave in numpy and found that your test does not provide the correct form of input it seems. https://numpy.org/doc/stable/reference/generated/numpy.lexsort.html

Reply:
Just to make sure, the ground truth value given by numpy's lexsort(np.lexsort([0, 0]) = 0) is the expected behavior.

Reply:
Wow I am stuck, I didn't expect this. I have a few scenarios that run well and these do not run the ivy tests. I have just committed another one and the error about "tuple index out of range". This is the all [test results of failed tests](https://drive.google.com/file/d/1a0kQwqESlMwhJe1_zFD6mNsC6bnUG_ze/view?usp=sharing). This is 23 pages document. I marked the out-of-range error part(pg 22).

Reply:
I know it looks like bad code and it is. According to my first commit about lexsort implementation in the torch backend, I thought it was clean and effective code to implement the function using the torch.unique but it failed. [torch.unique](https://pytorch.org/docs/stable/generated/torch.unique.html) returns unique elements in the tensor and can return reverse indices of them is optional. This time, I wanted to get reverse indices manually for loops. So I would save the function from the assertion error by only returning a value

Reply:
I am getting a "tuple index out of range" error while for example the test tensor equal to torch.Tensor([0]). I am confused because for lexsort it already must take multiple sorting keys, right?

Reply:
At this point, I think I have to make a decision. I was planning to complete torch.lexsort() and then tensorflow.lexsort(). What do you think should I continue to work on torch.lexsort() or should I move to the tensorflow.lexsort()?

Reply:
The lexsort requires a (k, N) array or tuple containing k (N,)-shaped sequences and should not work in other cases. I think this is the sourse of the out of range error you meet: 
![image](https://user-images.githubusercontent.com/45633544/205482204-06a6b77c-de09-4d54-8ec1-36627d58e0dd.png)

Reply:
You should first implement a way to check if the input have the correct shape by replicate what numpy will do in that case first.

Reply:
Now I get only one value from the torch.unique() function. 

I applied what you said and I found the exact reason for the error is numpy.lexsort() function returns for example: [0, 0] = [0], [1, 5, 9] = [0]. So if the input is a vector result will be 0.

In my function inverse_indices returns a tensor (same shape as input) representing the indices for where elements in the original input map to in the output. It means torch.argsort(inverse_indices, dim=axis)  returns [0, 0] = [1, 5, 9], [2, 1, 0]